app/assets/javascripts/templates.coffee

```diff
@@ -4,24 +4,67 @@ templateApp.factory "Template", ["$resource", ($resource) ->
   $resource("/templates/:id.json", {id: "@id"})
 ]
 
-templateApp.controller 'TemplateCtrl', ['$scope', 'Template', ($scope, Template) ->
+templateApp.controller 'TemplateCtrl', ['$scope', '$window', 'Template', ($scope, $window, Template) ->
   $scope.templates = Template.query()
 
   $scope.addTemplate = ->
+    t_name = $("#name").val()
+    if t_name.length == 0
+      alert "Name can't be blank"
+      return
+    found_template = $.grep($scope.templates, (e) ->
+      e.name == t_name
+    )
+    if found_template.length == 1
+      alert "This name already exist! " +
+          "Please, enter another name."
+      return
     template = Template.save($scope.newTemplate)
     $scope.templates.push(template)
     $scope.newTemplate = {}
+    $window.location.href = '/templates/index.html'
 
   $scope.removeTemplate = (int) ->
     int.$remove()
     $scope.templates.splice($scope.templates.indexOf(int), 1)
 ]
 
+templateApp.directive 'ngConfirmClick', [ ->
+  { link: (scope, element, attr) ->
+    msg = attr.ngConfirmClick or 'Are you sure?'
+    clickAction = attr.confirmedClick
+    element.bind 'click', (event) ->
+      if window.confirm(msg)
+        scope.$eval clickAction
+      return
+    return
+  }
+]
+
 templateApp.controller 'TemplateEditCtrl', ['$scope', '$window', 'Template', ($scope, $window, Template) ->
   $scope.template = Template.get({id: $(template_id)[0].value })
   $scope.templates = Template.query()
   $scope.editTemplate = (template) ->
-    template = Template.save($scope.template)
-    $scope.templates.push(template)
+    t_name = $("#name").val()
```

Comment:
you can not use jQuery in conjunction with Angular. These will can cause unexpected and hard to debug problems.
Find way to get needed value using Angular infrastructure.